host/bcm283x/clock.go

```diff
@@ -215,7 +215,7 @@ func calcSource(hz uint64, maxWaitCycles int) (clockCtl, int, int, uint64, error
 	if hz < 1 {
 		return 0, 0, 0, 0, fmt.Errorf("bcm283x-clock: desired frequency %dHz must be >1hz", hz)
 	}
-	if hz > 25000000 {
+	if hz > clk500MHz {
```

Comment:
I had selected 25Mhz because of:
- p. 105 about the MASH filter: "the user must ensure that the module
is not exposed to frequencies higher than 25MHz"

But the system can go higher than this, e.g.:
- p. 106 "The maximum operating frequency of the General Purpose clocks is ~125MHz at 1.2V but
this will be reduced if the GPIO pins are heavily loaded or have a capacitive load. "
- p. 21 "If the system clock is 250 MHz and the speed field is zero the SPI clock frequency is 125
MHz. The practical SPI clock will be lower as the I/O pads can not transmit or receive
signals at such high speed."

So I think choosing 125Mhz would be a good idea.

Reply:
Thanks for the info, 125MHz sounds reasonable. I also changed default PWM frequency to 25MHz.